test/Dialect/FIRRTL/emit-omir.mlir

```diff
@@ -471,6 +471,78 @@ firrtl.circuit "SRAMPathsWithNLA" attributes {annotations = [{
 // CHECK-SAME:   @MySRAM
 // CHECK-SAME: ]
 
+//===----------------------------------------------------------------------===//
+// Make SRAM Paths Absolute when SRAM is top-level (invalid for NLA)
+//===----------------------------------------------------------------------===//
+
+firrtl.circuit "SRAMPathsTopLevel" attributes {annotations = [{
+  class = "freechips.rocketchip.objectmodel.OMIRAnnotation",
+  nodes = [
+    {
+      info = #loc,
+      id = "OMID:0",
+      fields = {
+        omType = {info = #loc, index = 0, value = ["OMString:OMLazyModule", "OMString:OMSRAM"]},
+        finalPath = {info = #loc, index = 1, value = {omir.tracker, id = 0, type = "OMMemberReferenceTarget"}}
+      }
+    },
+    {
+      info = #loc,
+      id = "OMID:1",
+      fields = {
+        omType = {info = #loc, index = 0, value = ["OMString:OMLazyModule", "OMString:OMSRAM"]},
+        finalPath = {info = #loc, index = 1, value = {omir.tracker, id = 1, type = "OMMemberReferenceTarget"}}
+      }
+    }
+  ]
+}]} {
+  firrtl.extmodule @MySRAM()
+  firrtl.module @SRAMPathsTopLevel() {
+    firrtl.instance mem1 {annotations = [{class = "freechips.rocketchip.objectmodel.OMIRTracker", id = 0}]} @MySRAM()
+    %mem2_port = firrtl.mem Undefined {annotations = [{class = "freechips.rocketchip.objectmodel.OMIRTracker", id = 1}], depth = 8, name = "mem2", portNames = ["port"], readLatency = 0 : i32, writeLatency = 1 : i32 } : !firrtl.bundle<addr: uint<3>, en: uint<1>, clk: clock, data flip: uint<42>>
+  }
+}
+
+// CHECK-LABEL: firrtl.circuit "SRAMPathsTopLevel" {
+// CHECK-NOT: firrtl.hierpath
+// CHECK:         firrtl.extmodule @MySRAM()
+// CHECK-NEXT:    firrtl.module @SRAMPathsTopLevel() {
+// CHECK-NEXT:      firrtl.instance mem1 sym [[SYMMEM1:@[a-zA-Z0-9_]+]]
+// CHECK-SAME:        @MySRAM()
```

Comment:
This test can likely be golfed down to just a check of these two lines and checks of the path and the symbols.

Reply:
:+1:, thanks.  Done!